swap/src/protocol/alice.rs

```diff
@@ -39,29 +40,170 @@ pub mod state;
 mod steps;
 pub mod swap;
 
-pub type Swarm = libp2p::Swarm<Behaviour>;
+pub struct Swap {
```

Comment:
I wondering if it would be possible to change the DB to a reference here so we can support concurrent swaps (using the same DB).

Reply:
I would record that as 

> Approved but I think we should of attempted #131 first to gain some context on what the production abstractions should look like. I think they will change again once we try to support concurrent swaps.

I think the changes will fit on top of this just fine. I did think about concurrent swaps on the way, but I think we have to do several iterations anyway to level in on a suitable design. 
That said: I would merge this and then do changes on top. To my mind this is definitely an improvement to what was there before :)

Reply:
Fair point, I'd action this once we go for concurrent swaps.

Reply:
> I wondering if it would be possible to change the DB to a reference here so we can support concurrent swaps (using the same DB).

At this stage we still need to understand how the db can support concurrent swaps. No need to do this work with sled. Better migrate to sqlite first.